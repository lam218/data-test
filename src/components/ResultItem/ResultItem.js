//@flow
import * as React from "react";

import "./ResultItem.scss";

type Props = {
  item: Object,
  displayModal: () => void,
  row: Number,
};

const ResultItem = ({ item, displayModal, row }: Props) => (
  <div className="document" style={{ gridRow: row }}>
    <div className="document__content">
      <h2 className="document__title">{item.m_szDocTitle}</h2>
      <div className="document__meta">
        <div className="document__meta-date">
          <i className="far fa-calendar-alt"></i>
          {item.m_szYear}
        </div>
        <div className="document__meta-location">
          <i className="fas fa-map-marker-alt"></i>
          {item.m_szGeo1}
        </div>
      </div>
      <p>{item.m_szDocSumamry}</p>
      <div className="document__button-container">
        <button type="button" onClick={displayModal}>
          Click to expand
        </button>
      </div>
    </div>
  </div>
);

export default ResultItem;
