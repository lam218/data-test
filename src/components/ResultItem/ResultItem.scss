@import '../../styles/placeholders.scss';

.document {
  @extend %box-shadow;
  grid-column: 2 / 12;
  padding: ($basepx * 5);
  border-radius: $basepx;

  &__content {
    @extend %inner-container;
  }
  &__button-container {
    display: flex;
    justify-content: flex-end;
  }
  &__meta{
    display: flex;
  }
  &__meta-date{
    padding: 0 ($basepx * 2) 0 0;
  }
  &__meta-date, &__meta-location{
      i{
          padding: 0 $basepx 0 0;
      }
  }
}
