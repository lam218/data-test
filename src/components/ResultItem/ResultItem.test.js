import React from "react";
import renderer from "react-test-renderer";
import ResultItem from "./ResultItem";
import {  shallow, mount } from "enzyme";

test("Test that the Result item renders", () => {
  const displayModal = jest.fn();
  const component = renderer.create(
    <ResultItem
      row={1}
      item={{
        m_szDocTitle: "hello",
        m_szYear: "2020",
        m_szGeo1: "UK",
        m_szDocSumamry: "Some summary",
      }}
      displayModal={displayModal}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("Test that the Result item clicks", () => {
  const displayModal = jest.fn();
  const component = shallow(
    <ResultItem row={1} item={{ m_szDocTitle: "hello",
    m_szYear: "2020",
    m_szGeo1: "UK",
    m_szDocSumamry: "Some summary"}} displayModal={displayModal} />
  );
  component.find('button').simulate('click');
});
