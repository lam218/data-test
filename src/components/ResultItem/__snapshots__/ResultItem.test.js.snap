// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the Result item renders 1`] = `
<div
  className="document"
  style={
    Object {
      "gridRow": 1,
    }
  }
>
  <div
    className="document__content"
  >
    <h2
      className="document__title"
    >
      hello
    </h2>
    <div
      className="document__meta"
    >
      <div
        className="document__meta-date"
      >
        <i
          className="far fa-calendar-alt"
        />
        2020
      </div>
      <div
        className="document__meta-location"
      >
        <i
          className="fas fa-map-marker-alt"
        />
        UK
      </div>
    </div>
    <p>
      Some summary
    </p>
    <div
      className="document__button-container"
    >
      <button
        onClick={[MockFunction]}
        type="button"
      >
        Click to expand
      </button>
    </div>
  </div>
</div>
`;
